tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala

```diff
@@ -495,6 +524,31 @@ class TestQuintEx extends AnyFunSuite {
     assert(convert(Q.app("with", rec, Q.s, Q._42)(typ)) == """[["s" ↦ 1, "t" ↦ 2] EXCEPT ![<<"s">>] = 42]""")
   }
 
+  test("`with` operator conversion preserves row-typing") {
+    val typ = QuintRecordT.ofFieldTypes("a", ("s", QuintIntT()), ("t", QuintIntT()))
+    val rec = Q.app("Rec", Q.s, Q._1, Q.t, Q._2)(typ)
+    val exp = Q.quint.exToTla(rec).get
+    val expectedTlaType = RecRowT1(RowT1(VarT1("a"), ("s", IntT1), ("t", IntT1)))
+
+    assert(Quint.typeToTlaType(typ) == expectedTlaType)
+    assert(exp.typeTag == Typed(expectedTlaType))
+  }
+
+  test("operator def conversion preserves row-typing") {
+    // def updateF1(r) : {f1: int | a} => {s: int | a} = r.with("f1", 1)
```

Comment:
This signature is actually not type correct. I'm not sure if this is relevant for the test here. A simple way to make it correct is replacing `s` with `f1`, but perhaps you want something like this:
```
def updateF1(r) : { f1: int | a } => { s: int } = { s: r.f1 }
```

Reply:
Typo there. Thanks!